upgrade/steps-helpers.go

```diff
@@ -506,33 +505,26 @@ func getExpectedTargetFromIssues(ctx Context, name string) (*UpstreamUpgradeTarg
 			continue
 		}
 		v, err := semver.NewVersion(version)
-		if err == nil {
-			if ctx.InferVersion && !(ctx.TargetVersion == nil || ctx.TargetVersion.Equal(v) || ctx.TargetVersion.GreaterThan(v)) {
-				versionConstrained = true
-				continue
-			}
-			versions = append(versions, UpgradeTargetIssue{
-				Version: v,
-				Number:  title.Number,
-			})
+		if err != nil {
+			continue
 		}
+		versions = append(versions, UpgradeTargetIssue{
+			Version: v,
+			Number:  title.Number,
+		})
 	}
 	if len(versions) == 0 {
-		var extra string
-		if ctx.InferVersion && versionConstrained {
```

Comment:
is `ctx.InferVersion` used anywhere else? If not, should we get rid of it?

Reply:
We still use it: https://github.com/pulumi/upgrade-provider/blob/2d2e40c4b4cc0e849ee639331e95f22018ccbc56/upgrade/steps-helpers.go#L441.